Mathlib/Data/ZMod/Basic.lean

```diff
@@ -873,6 +920,27 @@ theorem neg_val {n : ℕ} [NeZero n] (a : ZMod n) : (-a).val = if a = 0 then 0 e
   rwa [le_zero_iff, val_eq_zero] at h
 #align zmod.neg_val ZMod.neg_val
 
+theorem val_neg_of_ne_zero {n : ℕ} [nz : NeZero n] (a : ZMod n) [na : NeZero a] :
+    (- a).val = n - a.val := by simp_all [neg_val a, na.out]
+
+theorem val_sub {n : ℕ} [NeZero n] {a b : ZMod n} (h : b.val ≤ a.val) :
+    (a - b).val = a.val - b.val := by
+  by_cases hb : b = 0
+  · cases hb; simp
+  · haveI : NeZero b := ⟨hb⟩
```

Comment:
```suggestion
  · have : NeZero b := ⟨hb⟩
```
haveI is inline have in lean 4, which you are unlikely to need, especially in a proof